### PR TITLE
docs(PPDSC-2204): release notes

### PR DESCRIPTION
**Pull request by mstuartf**
[PPDSC-2204](https://nidigitalsolutions.jira.com/browse/PPDSC-2204)

**What**

1. Background - provide info on our releases
2. What did you do - added a release notes page
3. What does the reviewers should expect - release notes page matching spec

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2204-release-notes
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2204-release-notes/storybook

File: site/utils/release-notes/functions.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    .reduce(
+      (prev: FullRelease[], next: Release) => [
+        ...prev,
+        {
+          ...next,
+          // Compare against the previous release in the array or the oldest
+          // release removed above.
+          change_level:
+            next.tag_name === 'v5.4.2' // todo: remove this (only here for design review)
+              ? 'major'
```
Comment (baburay23) [inline site/utils/release-notes/functions.ts:40]: does this todo still need to be here?

File: site/pages/about/release-notes.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import {GITHUB_URL, REPO} from '../../utils/release-notes/constants';
+import {IconFilledGitHub} from '../../components/icons/icon-filled-github';
+
+const ReleaseNotesPage = ({
+  path,
+  releases,
+  ...props
+}: LayoutProps & ReleasesPageProps) => {
+  const fullReleases = addChangeLevelToReleases(releases);
+  const [lastRelease, ...previousReleases] = fullReleases;
```
Comment (Vanals) [inline site/pages/about/release-notes.tsx:32]: Judging by the release notes: https://nidigitalsolutions.jira.com/wiki/spaces/NPP/pages/1334083669/Engineering+release+process
I think the latest note might not be showed correctly 🤔 . At the point, we are triggering the release, and everything gets built, in the GitHub `release` section we have a "draft note" with a title and a tag such as `trigger-release@x.x.x`. And unless we want to run to GitHub before the build happens and change the title and tag with the right ones, maybe the release page should not be static? if possible?

**Review by Vanals — CHANGES_REQUESTED** [review]

File: site/utils/release-notes/functions.ts (hunk 1 above)
Comment (mstuartf) [reply site/utils/release-notes/functions.ts:40]: Ah yes, I added that so there was a `major` release for design review. Will remove.

File: site/pages/about/release-notes.tsx (hunk 2 above)
Comment (mstuartf) [reply site/pages/about/release-notes.tsx:32]: Not sure I follow. What do you mean by 'immediately change the notes to github'?

Comment (mstuartf) [reply site/pages/about/release-notes.tsx:35]: Added a `updateFinalReleaseInfo` function to perform the update to the name, tag_name and compare link when required.

**Review by baburay23 — APPROVED** [review]

**Review by Vanals — APPROVED** [review]

**Pull request merged**